reference_entities/enrich_a_record_with_new_attribute_type.rst

```diff
@@ -0,0 +1,443 @@
+
+            $value = Value::create($attribute, $channelReference, $localeReference, $metricValue);
+            $record->setValue($value);
+        }
+    }
+
+Of course, we need to register this updater to be recognized by our registry:
+
+.. code-block:: yaml
+
+    acme.application.edit_record.record_value_updater.simple_metric_updater:
+            class: Acme\CustomBundle\Application\Record\EditRecord\ValueUpdater\SimpleMetricUpdater
+            tags:
+                - { name: akeneo_referenceentity.record_value_updater }
+
+
+3) Retrieve our record value
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Now that we can enrich our record with this new type of value, we need to create a dedicated hydrator, to hydrate our new record value from the DB:
+
+.. code-block:: php
+
+    <?php
+
+    namespace Acme\CustomBundle\Infrastructure\Persistence\Sql\Record\Hydrator;
+
+    use Acme\CustomBundle\Domain\Model\Attribute\SimpleMetricAttribute;
+    use Acme\CustomBundle\Domain\Model\Record\Value\SimpleMetricData;
+    use Akeneo\ReferenceEntity\Domain\Model\Attribute\AbstractAttribute;
+    use Akeneo\ReferenceEntity\Domain\Model\Record\Value\ValueDataInterface;
+    use Akeneo\ReferenceEntity\Infrastructure\Persistence\Sql\Record\Hydrator\DataHydratorInterface;
+
+    class SimpleMetricDataHydrator implements DataHydratorInterface
+    {
+        public function supports(AbstractAttribute $attribute): bool
+        {
+            return $attribute instanceof SimpleMetricAttribute;
+        }
+
+        public function hydrate($normalizedData): ValueDataInterface
+        {
+            return SimpleMetricData::createFromNormalize($normalizedData);
+        }
+    }
+
+And register it for the registry:
+
+.. code-block:: yaml
+
+    acme.infrastructure.persistence.record.hydrator.simple_metric_data:
+        class: Acme\CustomBundle\Infrastructure\Persistence\Sql\Record\Hydrator\SimpleMetricDataHydrator
+        tags:
+            - { name: akeneo_referenceentity.data_hydrator }
+
+Frontend Part of The New Record Value
+-------------------------------------
+
+To be able to enrich your records with this new attribute, we need to add some code in the frontend part.
+
+To do so, you can put all needed code in one single file but you can (and are encouraged) to split it into multiple
+files if needed.
+
+To keep this example simple, we will create everything in this file :
+
+``src/Acme/CustomBundle/Resources/public/reference-entity/record/simple-metric.tsx``
+
+.. note::
+
+    If you create a new Record Value, Akeneo will need three things to manage it in the frontend:
+
+    - A **model**: a representation of your Record Value, its properties and overall behaviour
+    - A **view**: as a React component to be able to render a user interface in the Record Form and dispatch events to the application
+    - A **cell**: as a React component to be able to render a cell in the Record Grid
```

Comment:
you could also add the explanation that everything can be done in one file and that it's the case for the example

Reply:
Already written just upper :)